### DISCUSSION TITLE
winxp binary system requirements? or intel cpu only?

### DISCUSSION THREAD

**Discussion by stanwebber** [FluidSynth/fluidsynth#1079] [Q&A] [answered]
i'm having trouble getting the latest (2.2.6) winxp binary working. i get a faulting module 'libfluidsynth-3.dll' in the application log after fluidsynth.exe immediately errors out when trying to run any command. i'm running xp media center edition sp3 with an athlon xp 2100. what are the software requirements to get this running? do the libraries use any sse2 instructions that would necessitate a p4 or later intel cpu?

**Comment by derselbst** [comment]
Yes, SSE2 could be the problem. I'm not specifying the `/arch` parameter for the WinXP build, in which case it seems to [default to SSE2](https://docs.microsoft.com/en-us/cpp/build/reference/arch-x86?view=msvc-170#arguments). This might also be the problem observed in #1077.

I tried specifying `/arch:IA32` on [this branch](https://github.com/FluidSynth/fluidsynth/tree/x87), but I'm getting a [bunch of errors](https://dev.azure.com/tommbrt/tommbrt/_build/results?buildId=7380&view=logs&j=ae601b3d-8756-52a4-4bcc-59c4d24c3f8c&t=1e42d489-b2c6-5e77-903e-6d797975357c). I need to have a deeper look into it. But I it'll take me another week before I can do so.

**Comment by stanwebber** [comment]
hope you can get it running with either ia32 or sse for all the p3/amd retro builds out there with isa soundcards. (early p4 chipsets can still technically do isa, but dma can be problematic.)

**Comment by derselbst** [comment, marked as answer]
I just found some time and got it compile. Pls. try the Win32 binaries here and let me know how it works for you:
https://dev.azure.com/tommbrt/tommbrt/_build/results?buildId=7408&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by stanwebber** [comment]
the libraries load now, but i'm having trouble hooking it up to midiyoke. no matter what i try it can't find the midi device. here's my fluidsynth command:
fluidsynth -o synth.polyphony=64 -a dsound -m winmidi -o midi.winmidi.device="In From MIDI Yoke:  1"  ..\..\sf2\8mbgm.sf2
i have painstakingly checked the midi device name even going so far as to cut & paste the text from midi-ox. am i doing something wrong or is it just broken?
also, i get an error about wasapi failing to initialize, but as i understand it, wasapi is vista or later only.

**Reply by derselbst** [reply]
Thanks for testing. When you run fluidsynth in the way you described, you can check the available midi devices by entering the command `info midi.winmidi.device` in fluidsynth's shell. This will report the available "options:" for this setting. Also, you could try running fluidsynth with the `-v` flag to potentially get some debug information.

**Comment by stanwebber** [comment]
thanks for the tip! the correct syntax ended up being "0: In From MIDI Yoke:  1".

as far as i can tell everything is working great now on my athlon xp 2100 system. is there anything else you'd like me to test? i hope you continue the precompiled xp binaries from this branch.

**Reply by derselbst** [reply]
> thanks for the tip! the correct syntax ended up being "0: In From MIDI Yoke: 1".

Ah yes, now I remember. That was due to a pretty recent change allowing to use multiple devices with this driver, in which case you would only need to specify the indices: https://www.fluidsynth.org/api/fluidsettings.xml#midi.winmidi.device

Glad to hear it works now! The binaries will be available with the 2.2.7 release again.
